### ISSUE TITLE
view cannot be created except in a module

### ISSUE THREAD

**Issue by danmackinlay** [tum-pbs/PhiFlow#67]
the following code cannot be executed as a cell in a jupyter notebook or python REPL:

```python
from phi.torch.flow import *
viewer = view()
```
We get an error in that case:
```text
AttributeError                            Traceback (most recent call last)
Input In [3], in <cell line: 37>()
     33     viewer.info(f"Model loaded.")
     36 prediction = CenteredGrid((0, 0), extrapolation.BOUNDARY, x=64, y=64)
---> 37 viewer = view('divergence', scene=True, namespace=globals(), select='batch')
     38 save_model(0)
     39 reset()  # Ensure that the first run will be identical to every time reset() is called

File ~/Source/fno_inversion_2022/venv/lib/python3.9/site-packages/phi/vis/_vis.py:138, in view(play, gui, name, description, scene, keep_alive, select, framerate, namespace, log_performance, *fields, **config)
    100 """
    101 Show `fields` in a graphical user interface.
    102 
   (...)
    133     `Viewer`
    134 """
    135 default_namespace = get_user_namespace(1)
    136 user_namespace = default_namespace if namespace is None else DictNamespace(namespace,
    137                                                                            title=default_namespace.get_title(),
--> 138                                                                            description=default_namespace.get_description(),
    139                                                                            reference=default_namespace.get_reference())
    140 variables = _default_field_variables(user_namespace, fields)
    141 actions = dict(ACTIONS)

File ~/Source/fno_inversion_2022/venv/lib/python3.9/site-packages/phi/vis/_user_namespace.py:175, in LocalNamespace.get_description(self)
    173     return self.function.__doc__
    174 else:
--> 175     return f"Function `{self.function_name}()` in '{self.module.__file__}'"

AttributeError: 'NoneType' object has no attribute '__file__'
```

Presumably this is because the  module is undefined when we are not in a module. Would it be of interest to replace `module.__file__` with something else (a hash?) in this case?

**Comment by holl-** [comment]
`view()` is currently not available in notebooks due to general performance issues and lack of resources for maintaining the UI. You can use `vis.plot()` instead to plot or animate your data, see the [animation gallery for 2.2](https://colab.research.google.com/github/tum-pbs/PhiFlow/blob/2.2-develop/docs/Animations.ipynb).

In the future, `view()` will probably use the same Dash interface as with regular scripts and display it inside the notebook.
You can already try this with
```python
!pip install dash

grid = CenteredGrid(Noise(), x=32, y=32)
view(grid, gui='dash', port=8051)
```

Then you have to display the web page one way or another which I havn't managed to get working with Google Colab yet.

**Comment by danmackinlay** [comment]
Interesting. Thanks for the update @holl- ; I am learning that I should be probably tracking the 2.2 branch. 
Side question: Would it be wise to track the 2.2-dev or 2.2.0 branches?

FWIW, interactive jupyter notebook graphics have never seemed like they were worth the effort to me (I'm not saying they are not valuable, but rather that the effort is very high); if I get my problem solved I'll try to submit a pull request to document your advice

**Comment by holl-** [comment]
`2.2-develop` gets the latest updates but may be unstable from time to time. Stable versions will be periodically merged into `2.0.0`. Once version `2.0.0` is released (planned for around July 27), the branch `2.0.0` will be merged into `master` and deleted.

> FWIW, interactive jupyter notebook graphics have never seemed like they were worth the effort to me (I'm not saying they are not valuable, but rather that the effort is very high)

Yeah, that's why `plot()` the main focus for notebooks. It makes displaying plots and animations very easy in notebooks!
